### PR TITLE
Feature/new profile screen

### PR DESCRIPTION
**Pull request by mattmeyer1066**
## What?
Added new profile creation screen for user onboarding with form validation and country search. Also changed location of theme tests.

## Why?
Users need a way to create their gardening profiles with personal info and experience level for personalized recommendations.

## How?
Built with Compose + Material 3. Features searchable country dropdown, experience selection, and form validation using MVVM pattern.

## Testing?
Added unit tests for ViewModel logic and UI tests for user interactions. Full coverage of form validation and dropdown behaviors.

### CODE REVIEW

File: app/src/main/java/com/android/mygarden/model/profile/Profile.kt
Comment (yngflibustier) [inline app/src/main/java/com/android/mygarden/model/profile/Profile.kt]: Documentation is missing for this enum class.

File: app/src/main/java/com/android/mygarden/ui/profile/NewProfileScreen.kt [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+ */
+@Composable
+private fun CountryDropdown(
+    uiState: NewProfileUIState,
+    newProfileViewModel: NewProfileViewModel,
+    isExpanded: Boolean,
+    onExpandedChange: (Boolean) -> Unit,
+    countryFocusRequester: FocusRequester
+) {
+  val filteredCountries =
```
Comment (yngflibustier) [inline app/src/main/java/com/android/mygarden/ui/profile/NewProfileScreen.kt]: Could the filtering part be done in the viewModel ? I think in this way the MVVM pattern is better respected.

File: app/src/main/java/com/android/mygarden/ui/profile/NewProfileScreen.kt [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+// Countries logic
+private const val MAX_COUNTRIES_DISPLAYED = 15
+
+/**
+ * Header section displaying the profile creation title and user avatar placeholder.
+ *
+ * @param modifier Modifier for customizing the layout
+ */
+@Composable
```
Comment (yngflibustier) [inline app/src/main/java/com/android/mygarden/ui/profile/NewProfileScreen.kt]: Profile Header LGTM.

File: app/src/main/java/com/android/mygarden/ui/profile/NewProfileViewModel.kt
Comment (yngflibustier) [inline app/src/main/java/com/android/mygarden/ui/profile/NewProfileViewModel.kt]: Does it really make sense to have 2 functions for each field ? Isn't the Error one only sufficient ?

File: app/src/androidTest/java/com/android/mygarden/ui/profile/NewProfileScreenTests.kt
Comment (yngflibustier) [inline app/src/androidTest/java/com/android/mygarden/ui/profile/NewProfileScreenTests.kt]: Nice Android tests ! You explained clearly what each test does, tested many inputs and handled edge cases !

File: app/src/androidTest/java/com/android/mygarden/ui/profile/NewProfileScreenTests.kt [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+  fun firstNameFieldAcceptsInput() {
+    setContent()
+
+    val testInput = "John"
+
+    // Enter text into the first name field
+    composeTestRule
+        .onNodeWithTag(NewProfileScreenTestTags.FIRST_NAME_FIELD)
+        .performTextInput(testInput)
+
```
Comment (yngflibustier) [inline app/src/androidTest/java/com/android/mygarden/ui/profile/NewProfileScreenTests.kt]: As the pattern "performTextInput" then "assertTextContains" is frequently repeated maybe create a helper function for the tests ! It will be easier to keep track of !

File: app/src/test/java/com/android/mygarden/ui/profile/NewProfileViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+    advanceUntilIdle()
+
+    val uiState = viewModel.uiState.value
+    assertFalse(uiState.canRegister())
+  }
+
+  @Test
+  fun canRegister_returnsFalseWhenLastNameEmpty() = runTest {
+    viewModel.setFirstName("John")
+    viewModel.setLastName("")
```
Comment (yngflibustier) [inline app/src/test/java/com/android/mygarden/ui/profile/NewProfileViewModelTest.kt]: As you repeat the same pattern of lines of code many times maybe try to do helper function for more consice code !

File: app/src/androidTest/java/com/android/mygarden/ui/profile/NewProfileScreenTests.kt (hunk 3 above)
Comment (mattmeyer1066) [reply app/src/androidTest/java/com/android/mygarden/ui/profile/NewProfileScreenTests.kt]: Good idea, I've made the change !

File: app/src/main/java/com/android/mygarden/model/profile/Countries.kt
[the remark this answers is not in the document]
Comment (mattmeyer1066) [reply app/src/main/java/com/android/mygarden/model/profile/Countries.kt]: Good idea for the API call, we'll discuss about it next meeting, I'm not sure if it is worth-it performance wise, but it is a great idea. For the xml we'll discuss it too. For now this was the simplest way to do it. I'll add a TODO.

File: app/src/main/java/com/android/mygarden/ui/profile/NewProfileViewModel.kt
Comment (yngflibustier) [reply app/src/main/java/com/android/mygarden/ui/profile/NewProfileViewModel.kt]: Oh, got it ! You are right !

**Review by yngflibustier — APPROVED** [review]
Thanks for addessing the changes quickly ! For me the code is now LGTM.
